core/src/main/java/hudson/model/AbstractBuild.java

```diff
@@ -459,49 +459,90 @@ public Result run(@Nonnull BuildListener listener) throws Exception {
+                        Set<LabelAtom> assignedLabels = new HashSet<>(node.getAssignedLabels());
+                        assignedLabels.remove(node.getSelfLabel());
+                        if (!assignedLabels.isEmpty()) {
+                            boolean first = true;
+                            for (LabelAtom label : assignedLabels) {
+                                if (first) {
+                                    listener.getLogger().print(" (");
+                                    first = false;
+                                } else {
+                                    listener.getLogger().print(' ');
+                                }
+                                listener.getLogger().print(label.getName());
                             }
-                            listener.getLogger().print(label.getName());
+                            listener.getLogger().print(')');
                         }
-                        listener.getLogger().print(')');
                     }
+                } else {
+                    listener.getLogger().print(Messages.AbstractBuild_Building());
                 }
-            } else {
-                listener.getLogger().print(Messages.AbstractBuild_Building());
-            }
-            
-            lease = decideWorkspace(node, Computer.currentComputer().getWorkspaceList());
+                
+                lease = decideWorkspace(node, Computer.currentComputer().getWorkspaceList());
 
-            workspace = lease.path.getRemote();
-            listener.getLogger().println(Messages.AbstractBuild_BuildingInWorkspace(workspace));
-            node.getFileSystemProvisioner().prepareWorkspace(AbstractBuild.this,lease.path,listener);
+                workspace = lease.path.getRemote();
+                listener.getLogger().println(Messages.AbstractBuild_BuildingInWorkspace(workspace));
+                node.getFileSystemProvisioner().prepareWorkspace(AbstractBuild.this,lease.path,listener);
 
-            for (WorkspaceListener wl : WorkspaceListener.all()) {
-                wl.beforeUse(AbstractBuild.this, lease.path, listener);
-            }
+                for (WorkspaceListener wl : WorkspaceListener.all()) {
+                    wl.beforeUse(AbstractBuild.this, lease.path, listener);
+                }
 
-            getProject().getScmCheckoutStrategy().preCheckout(AbstractBuild.this, launcher, this.listener);
-            getProject().getScmCheckoutStrategy().checkout(this);
+                getProject().getScmCheckoutStrategy().preCheckout(AbstractBuild.this, launcher, this.listener);
+                getProject().getScmCheckoutStrategy().checkout(this);
 
-            if (!preBuild(listener,project.getProperties()))
-                return Result.FAILURE;
+                if (!preBuild(listener,project.getProperties()))
```

Comment:
Maybe a good opportunity to include curly brackets here.

Reply:
If you ignore whitespaces, I'm not really modifying these lines of code:
https://github.com/jenkinsci/jenkins/pull/4517/files/15193af21d387218ed248182b2ef3e398ad0a593?utf8=%E2%9C%93&diff=split&w=1
So I'd rather not change the code style, I think it would be unrelated to the actual modifications from this PR.

Reply:
Fair enough!